ssl/ssl_lib.c

```diff
@@ -1341,6 +1344,23 @@ int SSL_set_fd(SSL *s, int fd)
     }
     BIO_set_fd(bio, fd, BIO_NOCLOSE);
     SSL_set_bio(s, bio, bio);
+#ifndef OPENSSL_NO_KTLS
+    {
+        /*
+         * The new socket is created successfully regardless of ktls_enable.
+         * ktls_enable doesn't change any functionality of the socket, except
+         * changing the setsockopt to enable the processing of ktls_start.
+         * Thus, it is not a problem to call it for non-TLS sockets.
+         */
+        int rc = ktls_enable(fd);
+
+        if (rc == 0) {
+#ifdef SSL_DEBUG
```

Comment:
Yuck. I'd prefer it if we don't add any more `SSL_DEBUG` stuff. `printf` just doesn't belong in the library.